test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+        /// -180° to 0° to +180° range then the assumption has to be made if the geometry closes towards the prime meridian or towards the dateline.
+        /// The current algorithm determines this by simply calculting which critical meridian is the closest.
+        /// https://github.com/NetTopologySuite/NetTopologySuite/discussions/496
+        /// https://github.com/NetTopologySuite/NetTopologySuite/blob/develop/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs
+        /// https://github.com/NetTopologySuite/NetTopologySuite/blob/84aef3c6a6cd3c2bfa5e3034ba161555cef5a117/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs#L24
+        /// https://github.com/nettopologysuite/nettopologysuite/issues/129
+        /// https://github.com/DotSpatial/DotSpatial/blob/e30421d6ee5b0cdf2b1fa01f95f8aabc50c1997a/Source/DotSpatial.Data/Shape.cs#L611
+        /// https://github.com/DotSpatial/DotSpatial/issues/1029
+        /// https://github.com/NetTopologySuite/ProjNet4GeoAPI/wiki/Well-Known-Text
+        /// https://desktop.arcgis.com/en/arcmap/10.3/manage-data/editing-fundamentals/creating-and-editing-multipart-polygons.htm
+        /// https://desktop.arcgis.com/en/arcmap/10.3/guide-books/map-projections/what-happens-to-features-at-180-dateline-.htm
+        /// </remarks>
+        public static Geometry ToPolyExOp(List<Coordinate> inpCoords, string inpProj4Wkt, OgcGeometryType outType, double outTrimGap, double outDensifyResolution, bool isAttFixOutInvPolygons)
+        {
+            Debug.Print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$");
+
+            // ###################################################################################################################
+            // input coordinate checks
+            // ###################################################################################################################
+            if (inpCoords == null)
+            {
+                throw new Exception("Null input coordinates.");
+            }
+
+            // check the minimum number of input coordinates for an output geometry
+            switch (outType)
+            {
+                case OgcGeometryType.LineString:
+                    if (inpCoords.Count < 2)
+                        throw new Exception("Input list has fewer than 2 coordinates. To create a polyline, the minimum is 2 coordinates.");
+                    break;
+                case OgcGeometryType.Polygon:
+                    if (inpCoords.Count < 3)
+                        throw new Exception("Input list has fewer than 3 coordinates. To create a polygon, the minimum is 3 coordinates.");
+                    break;
+                default:
+                    throw new Exception("Unsupported output geometry type.");
```

Comment:
NotSupportedException if the operation can not be implemented, or NotImplementedException if you plan to extend the code to support these operations in the future

Reply:
got it. thanks.